backend/app/usecase/validator/violation.go

```diff
@@ -0,0 +1,25 @@
+package validator
+
+// Violation represents a type of invalid error encountered by the validator.
+type Violation int
+
+const (
+	Valid Violation = iota
+	InvalidAlias
+	InvalidLongLink
```

Comment:
We should make those violation provide extract context to the API consumers. Not sure how will they react when seeing `InvalidAlias`

Reply:
What kind of context are you expecting? Should they see the custom alias they tried to enter?

Reply:
@Coteh I am not sure whether invalid long link a violation. Long link too long, long link empty are both violations. Invalid long link didn’t seem to communicate anything. The API consumer would have no clue how to handle it. Maybe we can consider removing violation code ( keep validation code but just not returning what is violating ) in the backend. What do you think?

Reply:
Are we able to replace those 2 lines with concrete errors?

Reply:
@byliuyang Sorry for late response. My original intention with `InvalidLongLink` and `InvalidAlias` were to be general catch-all error types for invalid errors that didn't have a concrete type defined yet. I understand your point and I think we can probably just return no "violation" error instead for these cases like you suggest.

Reply:
@byliuyang In the case that the long link or custom alias validation error contains no corresponding violation code, should the `extensions` field from the response contain no `violation` field? Would this be better than having `InvalidLongLink` and `InvalidAlias` as catch-all violation types? The benefit of omitting the `violation` field in this case is that the API consumer will only need to check whether the field is present in the response instead of checking for the catch-all types specifically. The downside is that it won't be 100% certain whether the `violation` field will be present in the error response.

Reply:
@Coteh How about this. Our use case will return the exact violation error so that we can track those metrics in DataDog. However, we API never return violation.

Reply:
Or we just complete remove the violation reason from the backend because frontend can already provide the user some hints on what's going wrong.